test/resource_test.rb

```diff
@@ -323,4 +494,56 @@ def destroy_resources
     end
     @resources = []
   end
+
+  # Utility function to test with multiple processes
+  # In particular, this is necessary to ensure handling of unique PIDs is correct,
+  # which is key to functionality that is utilized like SEM_UNDO
+  # Active workers are accumulated in the instance variable @workers,
+  # and workers must be cleaned up between tests by the teardown script
+  # An exit value of 100 is to keep track of timeouts, 0 for success.
+  def fork_workers(count:, resource: :testing, quota: nil, tickets: nil, timeout: 0.1, wait_for_timeout: false)
+    fail 'Must provide at least one of tickets or quota' unless tickets || quota
+
+    @workers ||= []
+    count.times do
+      @workers << fork do
+        begin
+          resource = Semian::Resource.new(resource.to_sym, quota: quota, tickets: tickets, timeout: timeout)
+          resource.acquire do
+            # Hold the resource until signalled
+            # This helps to avoid race conditions in testing.
+            Signal.trap('TERM') do
+              yield if block_given?
+              exit! 0
+            end
+            sleep
+          end
+        rescue Semian::TimeoutError
+          Signal.trap('TERM') do
```

Comment:
I believe I had a reason for this but I can't remember what it is or if it is still valid...